### ISSUE TITLE
Refactor `use-any` can remove required iterations of for loop

### ISSUE THREAD

**Issue by Gnome-py** [sourcery-ai/sourcery#146]
### Issue description or question
Below you can see an example of code sourcery wants to refactor:
```py
def must_be_called_for_all_items(item: str) -> bool:
    do_important_thing(item)
    return item == "all"

iterable = ["must", "go", "through", "all", "including", "a true item"]
all_in_list = False
for item in iterable:
    if must_be_called_for_all_items(item):
        all_in_list = True
```
Sourcery wants to refactor that loop to 
```py
iterable = ["must", "go", "through", "all", "including", "a true item"]
all_in_list = any(must_be_called_for_all_items(item) for item in iterable)
```
however this would cause `do_important_thing` to not be called for any item past `"all"`, I spotted this problem when jumping into the [python source code](https://github.com/python/cpython/blob/bce1418541a64a793960182772f985f64afbfa1a/Lib/asyncio/tasks.py#L670-L673) as in this example `child.cancel` would not be called for all items in the refactor `ret = any(child.cancel(msg=msg) for child in self._children)`.

### Sourcery Version
v0.9.3

### Code editor or IDE name and version

Visual Studio Code v1.58.0

### OS name and version
OS: Arch Linux
DE: KDE Plasma 5.22.3
Linux version: 5.12.15-arch1-1

**Comment by Hellebore** [comment]
Thanks for raising - will fix.

**Comment by ruancomelli** [comment]
This issue persists in v0.9.4. I just wanted to add that Sourcery seemingly is not aware of the need for a `break` statement.
The following two snippets are refactored the same way:
```py
all_in_list = False
for item in iterable:
    if must_be_called_for_all_items(item):
        all_in_list = True
```
and
```py
all_in_list = False
for item in iterable:
    if must_be_called_for_all_items(item):
        all_in_list = True
        break # note the break here
```
Both of them get refactored as `any(must_be_called_for_all_items(item) for item in iterable)`. The first one is wrong, but the second one is correct.

**Comment by Hellebore** [comment]
Fixed in 0.9.5
